operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -952,30 +957,57 @@ func getObservabilityAddon(c client.Client, namespace string,
 	}
 	err := c.Get(context.TODO(), namespacedName, found)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			return nil, nil
+		if !k8serrors.IsNotFound(err) {
+			log.Error(err, "Failed to check observabilityAddon")
+			return nil, err
 		}
-		log.Error(err, "Failed to check observabilityAddon")
-		return nil, err
 	}
 	if found.ObjectMeta.DeletionTimestamp != nil {
 		return nil, nil
 	}
-	return &mcov1beta1.ObservabilityAddon{
+
+	addon := &mcov1beta1.ObservabilityAddon{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: "observability.open-cluster-management.io/v1beta1",
 			Kind:       "ObservabilityAddon",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      obsAddonName,
-			Namespace: spokeNameSpace,
-		},
-		Spec: mcoshared.ObservabilityAddonSpec{
-			EnableMetrics: mco.Spec.ObservabilityAddonSpec.EnableMetrics,
-			Interval:      mco.Spec.ObservabilityAddonSpec.Interval,
-			Resources:     config.GetOBAResources(mco.Spec.ObservabilityAddonSpec, mco.Spec.InstanceSize),
+			Name:        obsAddonName,
+			Namespace:   spokeNameSpace,
+			Annotations: make(map[string]string),
 		},
-	}, nil
+	}
+
+	// Handle cases where the addon doesn't have the annotation
+	if found.Annotations == nil {
+		found.Annotations = make(map[string]string)
+	}
+
+	if _, ok := found.Annotations[addonSourceAnnotation]; !ok {
+		found.Annotations[addonSourceAnnotation] = "mco"
+	}
+
+	if found.Annotations[addonSourceAnnotation] == "mco" {
+		addon.Spec.EnableMetrics = mco.Spec.ObservabilityAddonSpec.EnableMetrics
+		addon.Spec.Interval = mco.Spec.ObservabilityAddonSpec.Interval
+		addon.Spec.ScrapeSizeLimitBytes = mco.Spec.ObservabilityAddonSpec.ScrapeSizeLimitBytes
+		addon.Spec.Workers = mco.Spec.ObservabilityAddonSpec.Workers
+		addon.Spec.Resources = config.GetOBAResources(mco.Spec.ObservabilityAddonSpec, mco.Spec.InstanceSize)
+
+		addon.Annotations[addonSourceAnnotation] = "mco"
+	}
+
+	if found.Annotations[addonSourceAnnotation] == "override" {
+		addon.Spec.EnableMetrics = found.Spec.EnableMetrics
+		addon.Spec.Interval = found.Spec.Interval
+		addon.Spec.ScrapeSizeLimitBytes = found.Spec.ScrapeSizeLimitBytes
+		addon.Spec.Workers = found.Spec.Workers
+		addon.Spec.Resources = found.Spec.Resources
+
+		addon.Annotations[addonSourceAnnotation] = "override"
```

Comment:
same comment as above? we wouldnt fall in here unless this k,v was already set?

Reply:
This is a new copy, but yeah we can just copy over from found.